### ISSUE TITLE
Negative Exponents of Ints treated as 0

### ISSUE THREAD

**Issue by CerebralSeed** [pytorch/pytorch#71633]
### 🐛 Describe the bug

Here is an example:

```
expo=torch.tensor([-1])
base=torch.tensor([10])
print(base**expo)

tensor([0])
```

### Versions

Version 1.9.0+cu111
Windows 10, 64 bit
Python 3.7

**Comment by gchanan** [comment]
I reproduced this on 1.10.1 linux.

**Comment by mruberry** [comment]
I think this is by design, although our docs could be improved.

There are a few ways this behavior could be implemented:

- `pow` could always return a floating point value to deal with negative exponents properly. This would be consistent with the behavior of other operations, like `sin`, where integer inputs return floating point results because the codomain of the integers includes non-integer real values. This would be a BC-breaking change and it maybe have numerical accuracy issues for large numbers.
- `pow`'s integer kernel(s) could test for negative exponents and throw an error on CPU. This would be consistent with NumPy. JAX will throw an error for scalar inputs (which PyTorch does not support) but return zero for array inputs like PyTorch does. If checks like these are performed they're typically CPU only. This would also be a BC-breaking change.
- `pow` could stay just like it is and be a little inconsistent with our other mathematical operations, and our documentation just needs to clarify this behavior.

Because the last one is not BC-breaking, consistent with JAX (where both JAX and PyTorch are defined) I think we just need to update our docs.

**Comment by CerebralSeed** [comment]
An easy work around is to just throw on a .float on the end. Anyways, figured it might be worth pointing out.
